client/ayon_resolve/api/lib.py

```diff
@@ -224,43 +223,58 @@ def create_bin(name: str,
 
 
 def remove_media_pool_item(media_pool_item: object) -> bool:
-    media_pool = get_current_project().GetMediaPool()
-    return media_pool.DeleteClips([media_pool_item])
+    """Remove media pool item.
 
+    Args:
+        media_pool_item (resolve.MediaPoolItem): resolve's object
 
-def create_media_pool_item(
-        files: list,
-        root: object = None,
-) -> object:
+    Returns:
+        bool: True if success
     """
-    Create media pool item.
+    resolve_project = get_current_resolve_project()
+    media_pool = resolve_project.GetMediaPool()
+    return media_pool.DeleteClips([media_pool_item])
+
+
+def create_media_pool_item(fpath: str,
+                           root: object = None) -> object:
+    """ Create media pool item.
 
     Args:
-        files (list[str]): list of absolute paths to files
+        fpath (str): absolute path to a file
         root (resolve.Folder)[optional]: root folder / bin object
 
     Returns:
         object: resolve.MediaPoolItem
     """
     # get all variables
-    media_pool = get_current_project().GetMediaPool()
+    media_storage = get_media_storage()
+    resolve_project = get_current_resolve_project()
+    media_pool = resolve_project.GetMediaPool()
     root_bin = root or media_pool.GetRootFolder()
 
-    # make sure files list is not empty and first available file exists
-    filepath = next((f for f in files if os.path.isfile(f)), None)
-    if not filepath:
-        raise FileNotFoundError("No file found in input files list")
-
     # try to search in bin if the clip does not exist
-    existing_mpi = get_media_pool_item(filepath, root_bin)
+    existing_mpi = get_media_pool_item(fpath, root_bin)
 
     if existing_mpi:
         return existing_mpi
 
-    # add all data in folder to media pool
-    media_pool_items = media_pool.ImportMedia(files)
+    dirname, file = os.path.split(fpath)
+    _name, ext = os.path.splitext(file)
 
-    return media_pool_items.pop() if media_pool_items else False
+    # add all data in folder to media-pool
+    media_pool_items = media_storage.AddItemListToMediaPool(
+        os.path.normpath(dirname))
```

Comment:
Can we make this work without doing this? I'd really prefer it to NOT load ALL media from the folder to the media pool.

Reply:
Yeah, I've restore current implementation. I think this changes were on initial draft PR but they do not seem relevant with this PR. I could confirm it all work successfully.